src/util/psbt/map/global.rs

```diff
@@ -154,6 +164,58 @@ impl Decodable for Global {
                                 return Err(Error::InvalidKey(pair.key).into())
                             }
                         }
+                        0x01 => {
+                            if !pair.key.key.is_empty() {
+                                let xpub = ExtendedPubKey::decode(&pair.key.key)
+                                    .map_err(|_| {
+                                        encode::Error::ParseFailed("Can't deserialize ExtendedPublicKey from global XPUB key data")
+                                    })?;
+                                let vlen: usize = pair.value.len();
+                                let mut decoder = Cursor::new(pair.value);
+                                let mut fingerprint = [0u8; 4];
+                                decoder.read_exact(&mut fingerprint[..])?;
+                                let mut path: Vec<ChildNumber> = Default::default();
```

Comment:
Hmm, about this whole section, it seems a bit strangely setup.

Basically after reading the fingerprint, the number of ChildNumbers is `decoder.len() / 4`, right? You can error early if `decoder.len() % 4 != 0` or even `pair.value.len() % 4 != 0` because the fingerprint is also 4 bytes.

And then `path` can be `Vec::with_capacity(decoder.len() / 4)` and then you can use while let for the parsing:
```rust
while let Ok(index) = u32::consensus_decode(&mut decoder) {
    path.push(index.into());
}
```

Reply:
Addressed with the new commit